apps/glusterfs/watcher.go

```diff
@@ -0,0 +1,92 @@
+//
+// Copyright (c) 2018 The heketi Authors
+//
+// This file is licensed to you under your choice of the GNU Lesser
+// General Public License, version 3 or any later version (LGPLv3 or
+// later), or the GNU General Public License, version 2 (GPLv2), in all
+// cases as published by the Free Software Foundation.
+//
+
+package glusterfs
+
+import (
+	"sync"
+	"time"
+
+	"github.com/boltdb/bolt"
+	"github.com/heketi/heketi/executors"
+	wdb "github.com/heketi/heketi/pkg/db"
+)
+
+//Do i really need to get read /write lock for this varaible?
+type nodesStatus struct {
+	Status map[string]bool
+	Mutex  sync.RWMutex
+}
+
+var NodesStatuses nodesStatus
+
+func init() {
+
+	NodesStatuses.Status = make(map[string]bool)
+}
+
+func NodeWatcher(db wdb.RODB, e executors.Executor) {
+	var status = make(map[string]bool)
+	err := db.View(func(tx *bolt.Tx) error {
+		nodes, err := NodeList(tx)
+		if err != nil {
+			return err
+		}
+		//to watch the node and update in memory
+
+		for _, n := range nodes {
+			if _, ok := status[n]; !ok {
+				status[n] = false
+			}
+			var newNode *NodeEntry
+			newNode, err = NewNodeEntryFromId(tx, n)
+
+			if err != nil {
+				//pass on to next node
+				continue
+
+			}
+
+			// should we check status of nodes disabled in db
+			//Ignore if the node is not online
+			if !newNode.isOnline() {
```

Comment:
I like this! My attempt was not checking this value but it makes sense not check on nodes that are administratively disabled.